config/git/src/main/kotlin/GitConfigFileProvider.kt

```diff
@@ -70,14 +70,17 @@ class GitConfigFileProvider internal constructor(
     constructor(gitUrl: String) : this(gitUrl, org.ossreviewtoolkit.utils.ort.createOrtTempDir())
 
     private val git = GitFactory.create()
+    private val lock = Any()
```

Comment:
Commit message:
- "only performed for test first time a config file is requested" not sure what this should say, maybe "for checking first time"?

Reply:
Why introduce `lock` instead of just doing `synchronized(this)` below?

Reply:
> It just felt a bit cleaner to have a dedicated lock object, but it doesn't make a big difference.

To me it's the opposite: If I read `synchronized(lock)`, I first need to convince myself about the lifetime of `lock` to understand the locking mechanism. But if like here the `lock` lifetime equals `this` lifetime, I strongly prefer the latter for clarity.

Reply:
I have added a commit on top.